### PR TITLE
Move native eligibility to the client

### PR DESCRIPTION
**Pull request by bluepnume**

### CODE REVIEW

File: src/api/eligibility.js [hunk 1]
```diff
@@ -71,3 +72,72 @@ export function getFundingEligibility(query : string, { accessToken, clientID, m
...
+        `,
+        variables: {
+            vault, shippingCallbackEnabled, merchantID, clientID,
+            buyerCountry, currency, userAgent, buttonSessionID, cookies
+        }
+    }).then((gqlResult) => {
+        if (!gqlResult || !gqlResult.mobileSDKEligibility) {
+            throw new Error(`GraphQL GetNativeEligibility returned no mobileSDKEligibility object`);
+        }
+        return gqlResult && gqlResult.mobileSDKEligibility;
```
Comment (mnicpt) [inline src/api/eligibility.js:141]: Is `gqlResult` check needed here?

File: src/payment-flows/native.js [hunk 2]
```diff
@@ -144,16 +145,15 @@ function isNativeEligible({ props, config, serviceData } : IsEligibleOptions) :
         return false;
     }
 
-    if (eligibility.nativeCheckout.paypal || eligibility.nativeCheckout.venmo) {
-        return true;
+    if (merchantID.length > 1) {
```
Comment (mnicpt) [inline src/payment-flows/native.js:148]: Why does this make them native ineligible?

Comment (abarco) [reply src/payment-flows/native.js:148]: This is to make MSP integrations ineligible for Native flows. I think the flow looks different for these integrations and Native is not ready to support them.

**Review by abarco — APPROVED** [review]

File: src/api/eligibility.js (hunk 1 above)
Comment (bluepnume) [reply src/api/eligibility.js:141]: Fair point 👍 removed

**Review by mnicpt — APPROVED** [review]

**Pull request merged**